### PR TITLE
AL/MIN-50. Integration tests liquidation.

### PR DESCRIPTION
**Pull request by lut1k**
**Description:**

1. Separated tests for runtime into separate modules.
2. Implemented an integration test for liquidation.

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [x] Code is covered with tests
- [x] Code is covered with benchmarks
- [x] Fresh benchmark results included

### CODE REVIEW

File: runtime/src/tests/balancing_pools_tests.rs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+				(BTC, Price::saturating_from_integer(50_000)),
+			];
+
+			MinterestOracle::on_finalize(0);
+
+			assert_ok!(MinterestOracle::feed_values(origin_of(ORACLE1::get().clone()), prices));
+
+			/*
+			Liquidity Pools balances (in assets): [2_700_000, 1_000_000, 2_500_000_000, 1_200_000]
+			Liquidity Pools balances (in USD): [81_000_000, 5_000_000, 3_750_000_000_000, 60_000_000_000]
```
Comment (DenisRomanovsky) [inline runtime/src/tests/balancing_pools_tests.rs:32]:
```suggestion
			Liquidation Pools balances (in USD): [81_000_000, 5_000_000, 3_750_000_000_000, 60_000_000_000]
```

File: runtime/src/tests/mod.rs [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+pub fn run_to_block(n: u32) {
+	while System::block_number() < n {
+		MinterestProtocol::on_finalize(System::block_number());
+		MinterestOracle::on_finalize(System::block_number());
+		System::set_block_number(System::block_number() + 1);
+	}
+}
+
+#[test]
+fn whitelist_mode_should_work() {
```
Comment (DenisRomanovsky) [inline runtime/src/tests/mod.rs:421]: You are moving tests to separate folders, so I'd suggest moving them all including the two remaining.

Comment (lut1k) [reply runtime/src/tests/mod.rs]: Fixed

File: runtime/src/tests/rpc_tests.rs [hunk 3]
```diff
@@ -0,0 +1,490 @@
+use super::*;
+
+#[test]
+fn demo_scenario_n2_without_insurance_using_rpc_should_work() {
```
Comment (dev-vadim) [inline runtime/src/tests/rpc_tests.rs:4]: Shouldn`t we change insurance to interest?

Comment (lut1k) [reply runtime/src/tests/rpc_tests.rs:4]: Fixed

**Review by Apollo50 — DISMISSED** [review]

**Review by kurotychb — APPROVED** [review]

File: runtime/src/tests/balancing_pools_tests.rs [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+}
+
+#[test]
+fn balance_liquidation_pools_two_pools_should_work_test() {
+	ExtBuilder::default()
+		.liquidity_pool_balance(DOT, 500_000 * DOLLARS)
+		.liquidity_pool_balance(ETH, 300_000 * DOLLARS)
+		.liquidation_pool_balance(DOT, 170_000 * DOLLARS) // + 140 000$
+		.liquidation_pool_balance(ETH, 30_000 * DOLLARS) //
+		// - 120 000$
```
[the remark this answers is not in the document]
Comment (lut1k) [reply runtime/src/tests/balancing_pools_tests.rs]: Fixed ><

**Review by kurotychb — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**